pcp/PCPDynamicColumn.c

```diff
@@ -0,0 +1,330 @@
+      if (trimmed[0] == '#' || trimmed[0] == '\0') {
+         free(trimmed);
+         continue;
+      }
+
+      size_t n;
+      char** config = String_split(trimmed, '=', &n);
+      free(trimmed);
+      if (config == NULL)
+         continue;
+
+      char* key = String_trim(config[0]);
+      char* value = n > 1 ? String_trim(config[1]) : NULL;
+      if (key[0] == '[') {  /* new section heading - i.e. new column */
+         PCPDynamicColumn_validateColumnName(key + 1, path, lineno);
+         column = PCPDynamicColumn_new(columns, key + 1);
+      } else if (value && String_eq(key, "caption")) {
+         free_and_xStrdup(&column->super.caption, value);
+      } else if (value && String_eq(key, "description")) {
+         free_and_xStrdup(&column->super.description, value);
+      } else if (value && String_eq(key, "width")) {
+         column->super.width = strtoul(value, NULL, 10);
+      } else if (value && String_eq(key, "maximum")) {
+         column->super.maximum = strtod(value, NULL);
+      } else if (value) {
+         PCPDynamicColumn_parseMetric(columns, column, path, lineno, key, value);
+      }
+      String_freeArray(config);
+      free(value);
+      free(key);
+   }
+   fclose(file);
+}
+
+static void PCPDynamicColumn_scanDir(PCPDynamicColumns* columns, char* path) {
+   DIR* dir = opendir(path);
+   if (!dir)
+      return;
+
+   struct dirent* dirent;
+   while ((dirent = readdir(dir)) != NULL) {
+      if (dirent->d_name[0] == '.')
+         continue;
+
+      char* file = String_cat(path, dirent->d_name);
+      PCPDynamicColumn_parseFile(columns, file);
+      free(file);
+   }
+   closedir(dir);
+}
+
+void PCPDynamicColumns_init(PCPDynamicColumns* columns) {
+   const char* sysconf = pmGetConfig("PCP_SYSCONF_DIR");
+   const char* xdgConfigHome = getenv("XDG_CONFIG_HOME");
+   const char* home = getenv("HOME");
+   char* path;
+
+   columns->table = Hashtable_new(0, true);
+
+   /* search in the users home directory first of all */
+   if (xdgConfigHome) {
+      path = String_cat(xdgConfigHome, "/htop/columns/");
+   } else {
+      if (!home)
+         home = "";
+      path = String_cat(home, "/.config/htop/columns/");
+   }
+   PCPDynamicColumn_scanDir(columns, path);
+   free(path);
+
+   /* secondly search in the system columns directory */
+   path = String_cat(sysconf, "/htop/columns/");
+   PCPDynamicColumn_scanDir(columns, path);
+   free(path);
+
+   /* check the working directory, as a final option */
+   char cwd[PATH_MAX];
+   if (getcwd(cwd, sizeof(cwd)) != NULL) {
+      path = String_cat(cwd, "/pcp/columns/");
+      PCPDynamicColumn_scanDir(columns, path);
+      free(path);
+   }
+}
+
+void PCPDynamicColumn_writeField(PCPDynamicColumn* this, const Process* proc, RichString* str, int field) {
+   const PCPProcess* pp = (const PCPProcess*) proc;
+   char buffer[30];
+   size_t size = sizeof(buffer);
+   int width = (abs(this->super.width) < 28 && this->super.width) ? this->super.width : -12;
```

Comment:
Switch order of conditions for consistency with the other location with this check in place.
```suggestion
   int width = (this->super.width && abs(this->super.width) < 28) ? this->super.width : -12;
```